README.md

```diff
@@ -18,28 +18,99 @@ In order to use the connector, you need a `SqsAsyncClient`. Refer to the [AWS SD
 
 ```scala
-def send(
+def make[R, T](
     client: SqsAsyncClient,
     queueUrl: String,
-    msg: String,
-    settings: SqsPublisherSettings = SqsPublisherSettings()
-  ): Task[Unit]
+    serializer: Serializer[T],
+    settings: ProducerSettings = ProducerSettings()
+  ): ZManaged[R with Clock, Throwable, Producer[T]]
 ```
 
-`SqsPublisherSettings` allows your to configure a number of things:
+where:
+
+- `client: SqsAsyncClient` - an instance of [SqsAsyncClient](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/sqs/SqsAsyncClient.html)
+- `queueUrl: String` - an SQS queue URL
+- `serializer: Serializer[T]` - a instance of [zio.sqs.serialization.Serializer](src/main/scala/zio/sqs/serialization/Serializer.scala) that can be used to convert an object of type `T` to a string.
+  ```scala
+    trait Serializer[T] {
+      def apply(t: T): String
+    }
+  ```
+  If the published message is already a string, `Serializer.serializeString` can be used.
+- `settings: ProducerSettings` - a set of options deifed in [ProducerSettings](src/main/scala/zio/sqs/producer/ProducerSettings.scala) and used to configure the queue
+  - `batchSize: Int` - The size of the batch to use, [1-10] (default: 10).
+  - `duration: Duration` - Time to wait for the batch to be full (have the specified batchSize) (default: 500 milliseconds).
+  - `parallelism: Int` - The number of concurrent requests to make to SQS (default: 16).
+  - `retryDelay: Duration` - If an event failed to be submit to SQS, retry it after the specified duration (default: 250 milliseconds).
+     The errors returned from SQS could either recoverable or not. An example of recoverable error -- when the server returned the code: `ServiceUnavailable`
+  - `retryMaxCount: Int` - The number of retried to make for the given event (default: 10)
+
+#### Producer
+
+[Producer](src/main/scala/zio/sqs/producer/Producer.scala) contains two sets of methods: with `E` suffix, e.g: `produceE`, `produceBatchE`, `sendStreamE` and the ones the suffix: `produce`, `produceBatch`, `sendStream` and `sendSink`.
+Methods with the suffix `E` return `ErrorOrEvent[T]` that is defied as `Either[ProducerError[T], ProducerEvent[T]]`, These methods do not fail the returned *Task* or *Stream* if SQS server returns an error for any of the published event. 
+Methods without `E` suffix fail the resulting *Task* or *Stream* if SQS server returns an error.
+
+`Producer` contains the following methods for event publishing:
+- `def produceE(e: ProducerEvent[T]): Task[ErrorOrEvent[T]]` - Publishes a single event and returns a Task that contains either the `ProducerError` or the `ProducerEvent` that was published.
+  Doesn't fail the `Task` if the server returns an error for the provided event.
+- `def produce(e: ProducerEvent[T]): Task[ProducerEvent[T]]` - Publishes a single event and fails the task.
```

Comment:
Could we maybe unify those 2 into `def produce(e: ProducerEvent[T]): IO[ProducerError, ProducerEvent[T]]`?

Reply:
will use Task[ProducerEvent[T]] instead, since `ProducerError` only set of SQS fails the message, but instead there could be other errors when the message batch is sent.